scripts/benchmark/checks/deploy_dos.py

```diff
@@ -0,0 +1,296 @@
+        super().__init__()
+
+        self.num_blocks = num_blocks
+        self.num_tx = num_tx
+
+        self.contract_interface = get_compiled_contract(
+            pathlib.Path(CONTRACT_FILE),
+            CONTRACT_NAME
+        )
+
+        self.w3 = Web3()
+
+    def _setup_benchmark(self, chain: MiningChain) -> None:
+        """
+        This hook can be overwritten to perform preparations on the chain
+        that do not count into the measured benchmark time
+        """
+        pass
+
+    @abstractmethod
+    def _apply_transaction(self, chain: MiningChain) -> None:
+        raise NotImplementedError(
+            "Must be implemented by subclasses"
+        )
+
+    def execute(self) -> DefaultStat:
+        total_stat = DefaultStat()
+
+        for chain in get_all_chains():
+
+            self._setup_benchmark(chain)
+
+            value = self.as_timed_result(
+                lambda: self.mine_blocks(chain, self.num_blocks, self.num_tx)
+            )
+
+            total_gas_used, total_num_tx = value.wrapped_value
+
+            stat = DefaultStat(
+                caption=chain.get_vm().fork,
+                total_blocks=self.num_blocks,
+                total_tx=total_num_tx,
+                total_seconds=value.duration,
+                total_gas=total_gas_used,
+            )
+            total_stat = total_stat.cumulate(stat)
+            self.print_stat_line(stat)
+
+        return total_stat
+
+    def mine_blocks(self, chain: MiningChain, num_blocks: int, num_tx: int) -> Tuple[int, int]:
+        total_gas_used = 0
+        total_num_tx = 0
+
+        for i in range(1, num_blocks + 1):
+            block = self.mine_block(chain, i, num_tx)
+            total_gas_used = total_gas_used + block.header.gas_used
+            total_num_tx = total_num_tx + len(block.transactions)
+
+        return total_gas_used, total_num_tx
+
+    def mine_block(self,
+                   chain: MiningChain,
+                   block_number: int,
+                   num_tx: int) -> BaseBlock:
+
+        for i in range(1, num_tx + 1):
+            self._apply_transaction(chain)
+
+        return chain.mine_block()
+
+    def deploy_dos_contract(self, chain: MiningChain) -> None:
+        # Instantiate the contract
+        dos_contract = self.w3.eth.contract(
+            abi=self.contract_interface['abi'],
+            bytecode=self.contract_interface['bin']
+        )
+
+        # Build transaction to deploy the contract
+        w3_tx1 = dos_contract.constructor().buildTransaction(W3_TX_DEFAULTS)
+
+        tx = new_transaction(
+            vm=chain.get_vm(),
+            private_key=FUNDED_ADDRESS_PRIVATE_KEY,
+            from_=FUNDED_ADDRESS,
+            to=CREATE_CONTRACT_ADDRESS,
+            amount=0,
+            gas=FIRST_TX_GAS_LIMIT,
+            data=decode_hex(w3_tx1['data']),
+        )
+
+        logging.debug('Applying Transaction {}'.format(tx))
+
+        block, receipt, computation = chain.apply_transaction(tx)
```

Comment:
doesn't look like we use either `block` or `receipt`.  Normally we use a placeholder variable of a single underscore for un-used variables like this.

```python
_, _, computation = chain.apply_transaction(tx)
```

Reply:
These placeholders don't work with mypy if both variables are of different types

Reply:
so I just left it as it was?